### PR TITLE
Improve RequiresCompiler performance

### PR DESCRIPTION
**Pull request by brasic**
### Motivation

When trying tapioca on a large codebase I noticed that the `require` subcommand would run for a very long time (more than 40 minutes).  When I profiled to investigate I found that almost all of time was spent running `File.basename` in a tight loop.

Looking closer, I found that most of [these calls](https://github.com/Shopify/tapioca/blob/fc1445a84a2f2287ae1c380140fb469d4ffc3a2f/lib/tapioca/compilers/requires_compiler.rb#L90) were redundant since `RequiresCompiler#name_in_project?` always receives the same `files` argument, so in all but the first call it is spending most of its time repeating previously done work.

### Implementation

I've inlined the helper method `RequiresCompiler#name_in_project?` and converted it to a set to speed up the large number of inclusion checks to be performed.  In the application I'm testing with this brings the total runtime of `tapioca require` from 40 minutes to 20 seconds.

### Tests

I believe this does not change behavior and is well covered by existing tests.  If you disagree I'm happy to add a test or benchmark of some kind.

### CODE REVIEW

**Review by KaanOzkan — APPROVED** [review]
Great find!

**Review by RyanBrushett — APPROVED** [review]

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: lib/tapioca/compilers/requires_compiler.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # frozen_string_literal: true
 
 require "spoom"
+require "set"
```
[the remark this answers is not in the document]
Comment (brasic) [reply lib/tapioca/compilers/requires_compiler.rb:5]: TIL!  I'll convert to a true-valued hash.

**Review by rafaelfranca — APPROVED** [review]

**Comment by brasic** [comment]
I just noticed that the CLA form is now working again, I've submitted the CLA now, can someone re-run the CLA task for me?  (I lack permission)

File: lib/tapioca/compilers/requires_compiler.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (paracycle) [reply lib/tapioca/compilers/requires_compiler.rb]: Hmm, we have a lot of usages of `set` in Tapioca already, so I guess we would need to add it to the gemspec anyway.

**Review by paracycle — APPROVED** [review]

**Pull request merged**